### PR TITLE
[Connector API][Docs] Document update error, last sync, name actions

### PR DESCRIPTION
**Pull request by jedrazb**
### Documentation
- Add docs for update_last_sync action
- Add docs for update_name action
- Add docs for update_error action

### CODE REVIEW

File: docs/reference/connector/apis/update-connector-last-sync-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+
+The value of `ConnectorSyncStatus` is one of the following lowercase strings representing different sync states:
+  - `canceling`: The sync process is in the process of being canceled.
+  - `canceled`: The sync process has been canceled.
+  - `completed`: The sync process completed successfully.
+  - `error`: An error occurred during the sync process.
+  - `in_progress`: The sync process is currently underway.
+  - `pending`: The sync is pending and has not yet started.
+  - `suspended`: The sync process has been temporarily suspended.
```
Comment (leemthompo) [inline docs/reference/connector/apis/update-connector-last-sync-api.asciidoc:73]:
```suggestion

* `canceling`: The sync process is in the process of being canceled.
* `canceled`: The sync process has been canceled.
* `completed`: The sync process completed successfully.
* `error`: An error occurred during the sync process.
* `in_progress`: The sync process is currently underway.
* `pending`: The sync is pending and has not yet started.
* `suspended`: The sync process has been temporarily suspended.
```

Comment (leemthompo) [reply docs/reference/connector/apis/update-connector-last-sync-api.asciidoc:73]: Updates to asciidoc syntax

**Review by leemthompo — APPROVED** [review]
LGTM 👍 , with one formatting fix for a list that isn't rendering correctly.

File: docs/reference/connector/apis/update-connector-last-sync-api.asciidoc (hunk 1 above)
Comment (jedrazb) [reply docs/reference/connector/apis/update-connector-last-sync-api.asciidoc:73]: Oh I'm still on a `.md` team it seems, thank you!

Comment (leemthompo) [reply docs/reference/connector/apis/update-connector-last-sync-api.asciidoc]: I do this at least once a day :)

**Comment by leemthompo** [comment]
@jedrazb FYI another backport fail here

**Comment by jedrazb** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.12|[](https://github.com/elastic/elasticsearch/pull/103528)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**